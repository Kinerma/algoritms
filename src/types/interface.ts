import {ElementStates} from "./element-states";
import {TStackContainer} from "../components/stack-page/stack";

export interface IString {
    state: ElementStates;
    letter: string
}

export interface IStackAdd {
    add: boolean;
    remove: boolean
}

export interface IStack<T> {
    push: (item: T) => void;
    pop: () => void;
    clear: () => void;
    elements: TStackContainer<T>
}

export interface ISorting {
    value: number;
    state: ElementStates
}


export interface IQueue<T> {
    enqueue: (item: T) => void;
    dequeue: () => void;
    clear: () => void;
    peak: () => T | undefined;
    getHead: () => number;
    getTail: () => number;
    getQueue: () => T[] | undefined[];
    getLength: () => number;
    getSize: () => number
}
