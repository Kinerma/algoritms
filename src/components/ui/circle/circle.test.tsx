import {Circle} from "./circle";
import renderer from "react-test-renderer";
import {ElementStates} from "../../../types/element-states";

describe('Компонент Circle', () => {
    it('Корректная отрисовка элемента без буквы', function () {
        const element = renderer.create(<Circle />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента с буквами', function () {
        const element = renderer.create(<Circle letter='text' />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента с head', function () {
        const element = renderer.create(<Circle head='head' />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка с react-элементом в head', function () {
        const element = renderer.create(<Circle head={<Circle />} />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента с tail', function () {
        const element = renderer.create(<Circle tail='tail' />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка с react-элементом в tail', function () {
        const element = renderer.create(<Circle tail={<Circle />} />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента с index', function () {
        const element = renderer.create(<Circle index={0} />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента с пропом isSmall === true', function () {
        const element = renderer.create(<Circle isSmall={true} />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента в состоянии default', function () {
        const element = renderer.create(<Circle state={ElementStates.Default} />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента в состоянии changing', function () {
        const element = renderer.create(<Circle state={ElementStates.Changing} />).toJSON()
        expect(element).toMatchSnapshot()
    });
    it('Корректная отрисовка элемента в состоянии modified', function () {
        const element = renderer.create(<Circle state={ElementStates.Modified} />).toJSON()
        expect(element).toMatchSnapshot()
    });
})