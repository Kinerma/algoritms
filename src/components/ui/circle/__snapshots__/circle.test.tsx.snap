// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Circle Корректная отрисовка с react-элементом в head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle   default"
        data-cy="border"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка с react-элементом в tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle   default"
        data-cy="border"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента без буквы 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента в состоянии changing 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   changing"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента в состоянии default 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента в состоянии modified 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   modified"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента с head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента с index 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента с tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="tail"
  >
    tail
  </div>
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента с буквами 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      text
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Компонент Circle Корректная отрисовка элемента с пропом isSmall === true 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  small default"
    data-cy="border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;
